### PR TITLE
[fio hack] imx: fiohab: ignore known HAB events

### PR DESCRIPTION
**Pull request by vanmaegima**
Patch based on work here:
https://git.pengutronix.de/cgit/barebox/commit/drivers/hab/habv4.c?id=620cd45ac1b97fb0f5cf68be309c6f45cc53cc6a

This works around a known HAB event generated by RNG failed
self-test.  We've added a fixed self-test into OP-TEE so it's
safe to ignore this event here for i.MX 6Q and i.MX 6DL.

Signed-off-by: Michael Scott <mike@foundries.io>
Signed-off-by: Vanessa Maegima <vanessa.maegima@foundries.io>

Please do not submit a Pull Request via github.  Our project makes use of
mailing lists for patch submission and review.  For more details please
see https://www.denx.de/wiki/U-Boot/Patches

### CODE REVIEW

File: arch/arm/mach-imx/fiohab.c [hunk 1]
```diff
@@ -3,6 +3,10 @@
  * Copyright (C) 2019 Foundries.IO
  */
 
+#if (defined(CONFIG_MX6Q) || defined(CONFIG_MX6DL) || defined(CONFIG_MX6QDL))
```
Comment (vanmaegima) [inline arch/arm/mach-imx/fiohab.c:6]: Added this check, the rest is all Mike's original patch.

**Comment by ricardosalveti** [comment]
Can you update the patch subject to use '[fio hack]' instead of HACK?

**Comment by vanmaegima** [comment]
> Can you update the patch subject to use '[fio hack]' instead of HACK?

@ricardosalveti done

**Comment by ricardosalveti** [comment]
@mike-scott @igoropaniuk mind having a look at this one?

**Review by Tim-Anderson — APPROVED** [review]
I yield to Mike but it LGTM

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Comment by igoropaniuk** [comment]
LGTM

**Pull request merged**